test/dotnet-add-proj.Tests/GivenDotnetAddProj.cs

```diff
@@ -0,0 +1,236 @@
+        {
+            var setup = CreateTestSetup(TestGroup, ProjectName);
+
+            var cmd = new AddProjCommand()
+                .WithWorkingDirectory(Path.Combine(setup.TestRoot, "MoreThanOne"))
+                .Execute($"\"{setup.GetProjectFullPath("Lib")}\"");
+            cmd.ExitCode.Should().NotBe(0);
+            cmd.StdErr.Should().Contain("more than one");
+            cmd.StdOut.Should().Contain("Usage");
+        }
+
+        [Fact]
+        public void WhenNoSolutionExistsInTheDirectoryItPrintsErrorAndUsage()
+        {
+            var setup = CreateTestSetup(TestGroup, ProjectName);
+
+            var cmd = new AddProjCommand()
+                .WithWorkingDirectory(Path.Combine(setup.TestRoot, "App"))
+                .Execute($"\"{setup.GetProjectFullPath("Lib")}\"");
+            cmd.ExitCode.Should().NotBe(0);
+            cmd.StdErr.Should().Contain("does not exist");
+            cmd.StdOut.Should().Contain("Usage");
+        }
+
+        [Theory]
+        [InlineData("Lib", "", false)]
+        [InlineData("LibWithProjectGuid", "{84A45D44-B677-492D-A6DA-B3A71135AB8E}", false)]
+        [InlineData("Lib", "", true)]
+        [InlineData("LibWithProjectGuid", "{84A45D44-B677-492D-A6DA-B3A71135AB8E}", true)]
+        public void ItAddsProjectAndPrintsStatus(string projectName, string projectGuid, bool makeRelative)
+        {
+            var setup = CreateTestSetup(TestGroup, ProjectName);
+            string slnDir = Path.Combine(setup.TestRoot, "MoreThanOne");
+            string slnName = "App.sln";
+
+            var projectToAdd = makeRelative
+                ? $"\"..{Path.DirectorySeparatorChar}{projectName}{Path.DirectorySeparatorChar}{projectName}.csproj\""
+                : $"\"{setup.GetProjectFullPath(projectName)}\"";
+
+            var cmd = new AddProjCommand()
+                .WithWorkingDirectory(slnDir)
+                .WithSolution(slnName)
+                .Execute(projectToAdd);
+            cmd.Should().Pass();
+            cmd.StdOut.Should().Contain("added to the solution");
+            cmd.StdErr.Should().BeEmpty();
+
+            var slnFile = new SlnFile();
+            slnFile.Read(Path.Combine(slnDir, slnName));
+            VerifyProjectInSolution(slnFile, projectName, projectGuid);
+
+            Restore(Path.Combine(setup.TestRoot, "App"), "App.csproj");
+            Build(slnDir, slnName);
+        }
+
+        [Fact]
+        public void WhenSolutionAlreadyContainsProjectItDoesntDuplicate()
+        {
+            var setup = CreateTestSetup(TestGroup, ProjectName);
+            string slnDir = Path.Combine(setup.TestRoot, "MoreThanOne");
+            string slnName = "AppAndLib.sln";
+            string slnFullPath = Path.Combine(slnDir, slnName);
+
+            var contentBefore = File.ReadAllText(slnFullPath);
+            var cmd = new AddProjCommand()
+                .WithWorkingDirectory(slnDir)
+                .WithSolution(slnName)
```

Comment:
We're using an absolute path in one place and a relative path in another. Is that meaningful to the test? Otherwise, it's extra cognitive load that we could eliminate...